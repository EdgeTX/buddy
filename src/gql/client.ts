import { ApolloClient, InMemoryCache } from "@apollo/client";

export default new ApolloClient({
  cache: new InMemoryCache(),
});
