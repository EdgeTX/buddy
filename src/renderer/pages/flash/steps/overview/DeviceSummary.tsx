import { UsbOutlined } from "@ant-design/icons";
import { Skeleton, Space, Typography } from "antd";
import React from "react";
import { Centered } from "renderer/pages/flash/shared";
import { Device } from "renderer/pages/flash/types";

const DeviceSummary: React.FC<{ device?: Device; loading?: boolean }> = ({
  device,
  loading,
}) => (
  <Space direction="vertical" size="large">
    <Centered>
      <UsbOutlined
        style={{
          fontSize: "60px",
          marginBottom: "8px",
        }}
      />
    </Centered>

    {loading ? (
      <Skeleton title active />
    ) : (
      <Centered>
        <Typography.Title style={{ textAlign: "center" }} level={4}>
          {device?.productName ?? "-"}
        </Typography.Title>
        <div>
          <Typography.Text style={{ textAlign: "center" }} type="secondary">
            {device?.serialNumber ?? "-"}
          </Typography.Text>
        </div>
        <div>
          <Typography.Text style={{ textAlign: "center" }} type="secondary">
            {device?.vendorId ?? "-"}:{device?.productId ?? "-"}
          </Typography.Text>
        </div>
      </Centered>
    )}
  </Space>
);

export default DeviceSummary;
