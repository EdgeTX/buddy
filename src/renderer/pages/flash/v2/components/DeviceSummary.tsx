import { UsbOutlined } from "@ant-design/icons";
import { Space, Typography } from "antd";
import React from "react";
import { Centered } from "renderer/pages/flash/v2/shared";
import { Device } from "renderer/pages/flash/v2/types";

const DeviceSummary: React.FC<{ device: Device }> = ({ device }) => (
  <Space direction="horizontal" size="large">
    <Centered>
      <UsbOutlined
        style={{
          fontSize: "64px",
        }}
      />
    </Centered>
    <div>
      <Typography.Title level={4}>{device.productName}</Typography.Title>
      <div>
        <Typography.Text type="secondary">
          {device.serialNumber ?? "-"}
        </Typography.Text>
      </div>
      <div>
        <Typography.Text type="secondary">
          {device.vendorId}:{device.productId}
        </Typography.Text>
      </div>
    </div>
  </Space>
);

export default DeviceSummary;
