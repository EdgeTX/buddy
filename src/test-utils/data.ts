export const exampleReleasesList = [
  {
    id: "nightly",
    name: "nightly",
    isPrerelease: true,
  },
  {
    id: "v2.5.0",
    name: 'EdgeTX "Dauntless" 2.5.0',
    isPrerelease: false,
  },
  {
    id: "v2.5.0-rc3",
    name: 'EdgeTX "Dauntless" 2.5.0-RC3',
    isPrerelease: true,
  },
  {
    id: "v2.5.0-rc2",
    name: 'EdgeTX "Dauntless" 2.5.0-RC2',
    isPrerelease: true,
  },
  {
    id: "v2.5.0-rc1",
    name: 'EdgeTX "Dauntless" 2.5.0-RC1',
    isPrerelease: true,
  },
  {
    id: "v2.4.0",
    name: 'EdgeTX "Endeavour" 2.4.0',
    isPrerelease: false,
  },
  {
    id: "v2.4.0-rc4",
    name: "EdgeTX Endeavour (2.4.0) RC4",
    isPrerelease: true,
  },
  {
    id: "v2.4.0-rc3",
    name: "EdgeTX Endeavour (2.4.0) RC3",
    isPrerelease: true,
  },
  {
    id: "v2.4.0-rc2",
    name: "EdgeTX Endeavour (2.4.0) RC2",
    isPrerelease: true,
  },
  {
    id: "v2.4.0-rc1",
    name: "2.4.0-rc1",
    isPrerelease: true,
  },
];

export const exampleTargetsList = [
  {
    code: "nv14",
    id: "nv14-50406116",
    name: "Flysky NV14",
  },
  {
    code: "x10",
    id: "x10-50406116",
    name: "Frsky Horus X10",
  },
  {
    code: "x10-access",
    id: "x10-access-50406116",
    name: "Frsky Horus X10 Access",
  },
  {
    code: "x12s",
    id: "x12s-50406116",
    name: "Frsky Horus X12s",
  },
  {
    code: "x7",
    id: "x7-50406116",
    name: "Frsky QX7",
  },
  {
    code: "x7-access",
    id: "x7-access-50406116",
    name: "Frsky QX7 Access",
  },
  {
    code: "x9d",
    id: "x9d-50406116",
    name: "Frsky X9D",
  },
  {
    code: "x9dp",
    id: "x9dp-50406116",
    name: "Frsky X9D Plus",
  },
  {
    code: "x9dp2019",
    id: "x9dp2019-50406116",
    name: "Frsky X9D Plus 2019",
  },
  {
    code: "x9lite",
    id: "x9lite-50406116",
    name: "Frsky X9 Lite",
  },
  {
    code: "x9lites",
    id: "x9lites-50406116",
    name: "Frsky X9 Lite S",
  },
  {
    code: "xlite",
    id: "xlite-50406116",
    name: "Frsky X-Lite",
  },
  {
    code: "xlites",
    id: "xlites-50406116",
    name: "Frsky X-Lite S",
  },
  {
    code: "t8",
    id: "t8-50406116",
    name: "Jumper T8",
  },
  {
    code: "t12",
    id: "t12-50406116",
    name: "Jumper T12",
  },
  {
    code: "t16",
    id: "t16-50406116",
    name: "Jumper T16",
  },
  {
    code: "t18",
    id: "t18-50406116",
    name: "Jumper T18",
  },
  {
    code: "tlite",
    id: "tlite-50406116",
    name: "Jumper T-Lite",
  },
  {
    code: "tx12",
    id: "tx12-50406116",
    name: "Radiomaster TX12",
  },
  {
    code: "tx16s",
    id: "tx16s-50406116",
    name: "Radiomaster TX16s",
  },
];

export const exampleDevices = [
  {
    id: "1",
    serialNumber: "375897983475",
    productName: "STEM32 Bootloader",
    vendorId: "0x3453",
    productId: "0x4358",
  },
  {
    id: "2",
    serialNumber: "2435883",
    productName: "Radio 2",
    vendorId: "0x5467",
    productId: "0x4668",
  },
  {
    id: "3",
    serialNumber: "2435883",
    productName: "Radio 3",
    vendorId: "0x5467",
    productId: "0x4668",
  },
  {
    id: "4",
    serialNumber: "2435883",
    productName: "Radio 4",
    vendorId: "0x5467",
    productId: "0x4668",
  },
  {
    id: "5",
    serialNumber: "375897983475",
    productName: "STEM32 Bootloader 2",
    vendorId: "0x3453",
    productId: "0x4358",
  },
  {
    id: "6",
    serialNumber: "375897983475",
    productName: "STEM32 Bootloader 3",
    vendorId: "0x3453",
    productId: "0x4358",
  },
  {
    id: "6",
    serialNumber: "375897983475",
    productName: "STEM32 Bootloader 4",
    vendorId: "0x3453",
    productId: "0x4358",
  },
];

export const examplePrs = [
  {
    headCommitId: "217c02e6e06b4500edbb0eca99b5d1d077111aab",
    id: "1333",
    name: "wimalopaan:wm1333",
  },
  {
    headCommitId: "2fceb40b70432378d9e0693e0d3939221799ba86",
    id: "984",
    name: "EdgeTX:crsf-support-lua-command-id",
  },
  {
    headCommitId: "621a6833106e744b7e4b616ad6218af7c7d47e19",
    id: "1009",
    name: "jfrickmann:LCD",
  },
  {
    headCommitId: "ceafdea344cd9bc021dd083fe4a487393edb1458",
    id: "980",
    name: "EdgeTX:nv14-bigger-controls",
  },
  {
    headCommitId: "c05c7066af63dfd30df28fbf1355ef19eca0f574",
    id: "1115",
    name: "stronnag:add_MP_header_to_Telem_Mirror",
  },
];

export const examplePrCommits = [
  {
    id: "217c02e6e06b4500edbb0eca99b5d1d077111aab",
  },
  {
    id: "b9319c5428aab118b9473e98fcde996355333c8c",
  },
];

export const exampleCloudbuildTargets = {
  releases: {
    nightly: {
      sha: "06712d9b977ee19170a3abc3c183b2f17bd8fa35",
      remote: {
        url: "https://api.github.com/repos/EdgeTX/edgetx/git/ref/tags/nightly",
        json_path: "",
      },
    },
    "v2.8.4": {
      sha: "da28e356449e54c57f0e5e356bd5ec5709128ff7",
      remote: null,
      exclude_targets: ["tlitef4"],
    },
  },
  flags: {
    language: {
      build_flag: "TRANSLATIONS",
      values: [
        "EN",
        "CZ",
        "DA",
        "DE",
        "EN",
        "ES",
        "FI",
        "FR",
        "IT",
        "PT",
        "SK",
        "SE",
        "PL",
        "HU",
        "NL",
      ],
    },
  },
  tags: { colorlcd: { flags: { language: { values: ["CN", "JP", "TW"] } } } },
  targets: {
    boxer: {
      description: "RadioMaster Boxer",
      build_flags: { PCB: "X7", PCBREV: "BOXER" },
    },
    commando8: {
      description: "iFlight Commando 8",
      build_flags: { PCB: "X7", PCBREV: "COMMANDO8" },
    },
    el18: {
      description: "Flysky EL18",
      tags: ["colorlcd"],
      build_flags: { PCB: "NV14" },
    },
    lr3pro: {
      description: "BETAFPV LiteRadio 3 Pro",
      build_flags: { PCB: "X7", PCBREV: "LR3PRO" },
    },
    nv14: {
      description: "Flysky NV14",
      tags: ["colorlcd"],
      build_flags: { PCB: "NV14" },
    },
    t12: {
      description: "Jumper T12",
      build_flags: { PCB: "X7", PCBREV: "T12" },
    },
    t16: {
      description: "Jumper T16",
      tags: ["colorlcd"],
      build_flags: { PCB: "X10", PCBREV: "T16" },
    },
    t18: {
      description: "Jumper T18",
      tags: ["colorlcd"],
      build_flags: { PCB: "X10", PCBREV: "T18" },
    },
    t8: {
      description: "Radiomaster T8",
      build_flags: { PCB: "X7", PCBREV: "T8" },
    },
    tlite: {
      description: "Jumper T-Lite",
      build_flags: { PCB: "X7", PCBREV: "TLITE" },
    },
    tlitef4: {
      description: "Jumper T-Lite (F4 MCU)",
      build_flags: { PCB: "X7", PCBREV: "TLITEF4" },
    },
    tpro: {
      description: "Jumper T-Pro",
      build_flags: { PCB: "X7", PCBREV: "TPRO" },
    },
    tx12: {
      description: "Radiomaster TX12MK2",
      build_flags: { PCB: "X7", PCBREV: "TX12MK2" },
    },
    tx16s: {
      description: "RadioMaster TX16S",
      tags: ["colorlcd"],
      build_flags: { PCB: "X10", PCBREV: "TX16S" },
    },
    x10: {
      description: "FrSky Horus X10",
      tags: ["colorlcd"],
      build_flags: { PCB: "X10" },
    },
    "x10-access": {
      description: "FrSky Horus X10 Access",
      tags: ["colorlcd"],
      build_flags: { PCB: "X10", PCBREV: "EXPRESS", PXX1: "YES" },
    },
    x12s: {
      description: "FrSky Horus X12S",
      tags: ["colorlcd"],
      build_flags: { PCB: "X12S" },
    },
    x7: { description: "FrSky QX7", build_flags: { PCB: "X7" } },
    "x7-access": {
      description: "FrSky QX7 Access",
      build_flags: { PCB: "X7", PCBREV: "ACCESS" },
    },
    x9d: { description: "FrSky X9D", build_flags: { PCB: "X9D" } },
    x9dp: {
      description: "FrSky X9D Plus",
      build_flags: { PCB: "X9D+", PCBREV: "ACCESS" },
    },
    x9dp2019: {
      description: "FrSky X9D Plus 2019",
      build_flags: { PCB: "X9D+", PCBREV: "2019" },
    },
    x9e: { description: "FrSky X9E", build_flags: { PCB: "X9E" } },
    x9lite: { description: "FrSky X9 Lite", build_flags: { PCB: "X9LITE" } },
    x9lites: {
      description: "FrSky X9 Lite S",
      build_flags: { PCB: "X9LITES" },
    },
    xlite: { description: "FrSky X-Lite", build_flags: { PCB: "XLITE" } },
    xlites: { description: "FrSky X-Lite S", build_flags: { PCB: "XLITES" } },
    zorro: {
      description: "RadioMaster Zorro",
      build_flags: { PCB: "X7", PCBREV: "ZORRO" },
    },
  },
};
