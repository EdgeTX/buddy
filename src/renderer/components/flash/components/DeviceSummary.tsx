import { UsbOutlined } from "@ant-design/icons";
import { Skeleton, Space, Typography } from "antd";
import React from "react";
import { Centered } from "renderer/shared/layouts";
import { Device } from "renderer/components/flash/types";
import { gql, useQuery } from "@apollo/client";

const DeviceDetails: React.FC<{ device?: Device; loading?: boolean }> = ({
  device,
  loading,
}) => (
  <Space direction="vertical" size="large" style={{ width: "100%" }}>
    <Centered>
      <UsbOutlined
        style={{
          fontSize: "60px",
          marginBottom: "8px",
        }}
      />
    </Centered>

    {loading ? (
      <Skeleton title paragraph={{ rows: 2 }} active />
    ) : (
      <Centered>
        <Typography.Title style={{ textAlign: "center" }} level={5}>
          {device?.productName ?? "-"}
        </Typography.Title>
        <div>
          <Typography.Text style={{ textAlign: "center" }} type="secondary">
            {device?.serialNumber ?? "-"}
          </Typography.Text>
        </div>
        <div>
          <Typography.Text style={{ textAlign: "center" }} type="secondary">
            {device?.vendorId ?? "-"}:{device?.productId ?? "-"}
          </Typography.Text>
        </div>
      </Centered>
    )}
  </Space>
);

const DeviceSummary: React.FC<{ deviceId: string }> = ({ deviceId }) => {
  const { loading, data } = useQuery(
    gql(/* GraphQL */ `
      query DeviceInfo($deviceId: ID!) {
        flashableDevice(id: $deviceId) {
          id
          productName
          serialNumber
          vendorId
          productId
        }
      }
    `),
    {
      variables: {
        deviceId,
      },
    }
  );

  return (
    <DeviceDetails
      loading={loading}
      device={data?.flashableDevice ?? undefined}
    />
  );
};

export default DeviceSummary;
