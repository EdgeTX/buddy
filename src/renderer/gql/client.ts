import { ApolloClient, InMemoryCache } from "@apollo/client";
import { createWebWorkerBusLink } from "apollo-bus-link/webworker";
import worker from "../../webworker/webworker.bootstrap";

const link = createWebWorkerBusLink(worker);
link.initialiseBackend({});

export default new ApolloClient({
  cache: new InMemoryCache(),
  link,
});
