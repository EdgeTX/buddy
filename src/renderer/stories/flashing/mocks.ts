import { MockedResponse } from "@apollo/client/testing";
import gql from "graphql-tag";
import { times } from "shared/tools";
import {
  exampleDevices,
  exampleReleasesList,
  exampleTargetsList,
} from "test-utils/data";

export const firmwaresQuery: MockedResponse = {
  request: {
    query: gql`
      query Releases {
        edgeTxReleases {
          id
          name
          isPrerelease
        }
      }
    `,
  },
  result: {
    data: {
      edgeTxReleases: exampleReleasesList,
    },
  },
  delay: 100,
};

export const targetsQuery: MockedResponse = {
  request: {
    query: gql`
      query ReleaseTargets($releaseId: ID!) {
        edgeTxRelease(id: $releaseId) {
          id
          firmwareBundle {
            id
            targets {
              id
              name
            }
          }
        }
      }
    `,
    variables: {
      releaseId: "v2.5.0",
    },
  },
  result: {
    data: {
      edgeTxRelease: {
        id: "",
        firmwareBundle: {
          id: "",
          targets: exampleTargetsList,
        },
      },
    },
  },
  delay: 1000,
};

const releaseDescription = times(10)
  .map(
    () => `
My release, it is great, it has lots of things.

- This thing
- and this
- and this.

[View it here](https://google.com)
`
  )
  .join("");

export const firmwareReleaseDescriptionQuery: MockedResponse = {
  request: {
    query: gql`
      query FirmwareReleaseDescription($releaseId: ID!) {
        edgeTxRelease(id: $releaseId) {
          id
          description
        }
      }
    `,
    variables: {
      releaseId: "v2.5.0",
    },
  },
  delay: 200,
  result: {
    data: {
      edgeTxRelease: {
        id: "v2.5.0",
        description: releaseDescription,
      },
    },
  },
};

export const devicesQuery: MockedResponse = {
  request: {
    query: gql`
      query Devices {
        flashableDevices {
          id
          productName
          serialNumber
          vendorId
          productId
        }
      }
    `,
  },
  delay: 200,
  result: {
    data: {
      flashableDevices: exampleDevices,
    },
  },
};
