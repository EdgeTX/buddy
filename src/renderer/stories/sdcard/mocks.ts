import { MockedResponse } from "@apollo/client/testing";
import gql from "graphql-tag";
import { exampleTargetsList } from "test-utils/data";

// eslint-disable-next-line import/prefer-default-export
export const pickSdcardDirectoryMutation = (id?: string): MockedResponse => ({
  request: {
    query: gql`
      mutation PickSdcardDirectory {
        pickSdcardDirectory {
          id
        }
      }
    `,
  },
  result: {
    data: {
      pickSdcardDirectory: id
        ? {
            __typename: "SdcardDirectory",
            id,
          }
        : null,
    },
  },
});

export const sdcardInfoQuery = (
  directoryId: string,
  details?: { isValid: boolean; version?: string; target?: string }
): MockedResponse => ({
  request: {
    query: gql`
      query SdcardInfo($directoryId: ID!) {
        sdcardDirectory(id: $directoryId) {
          id
          isValid
          version
          target
        }
      }
    `,
    variables: {
      directoryId,
    },
  },
  result: {
    data: {
      sdcardDirectory: details
        ? {
            __typename: "SdcardDirectory",
            id: directoryId,
            target: null,
            version: null,
            ...details,
          }
        : null,
    },
  },
});

export const sdcardAssetInfoQuery = (
  directoryId: string,
  details: { version?: string; target?: string; sounds: string[] }
): MockedResponse => ({
  request: {
    query: gql`
      query SdcardAssetInfo($directoryId: ID!) {
        sdcardDirectory(id: $directoryId) {
          id
          name
          version
          target
          sounds
        }
      }
    `,
    variables: {
      directoryId,
    },
  },
  result: {
    data: {
      sdcardDirectory: {
        __typename: "SdcardDirectory",
        id: directoryId,
        target: null,
        version: null,
        ...details,
      },
    },
  },
});

export const sdcardPacksQuery: MockedResponse = {
  request: {
    query: gql`
      query SdcardPacks {
        edgeTxSdcardPackReleases {
          id
          name
          isPrerelease
          targets {
            id
            name
          }
        }
      }
    `,
  },
  result: {
    data: {
      edgeTxSdcardPackReleases: [
        {
          id: "v2.5.0",
          name: "v2.5.0",
          isPrerelease: false,
          targets: exampleTargetsList,
        },
      ],
    },
  },
};
