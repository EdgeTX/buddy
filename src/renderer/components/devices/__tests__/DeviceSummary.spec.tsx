import { MockedProvider } from "@apollo/client/testing";
import React from "react";
import { render } from "test-utils/testing-library";
import DeviceSummary from "renderer/components/devices/DeviceSummary";
import { deviceQuery } from "test-utils/mocks";
import { exampleDevices } from "test-utils/data";
import { screen } from "@testing-library/react";
import snapshotDiff from "snapshot-diff";

const device = exampleDevices[4]!;

describe("<DeviceSummary />", () => {
  it("should show details about the given device", async () => {
    const { asFragment } = render(
      <MockedProvider mocks={[deviceQuery(0)]}>
        <DeviceSummary deviceId="1" />
      </MockedProvider>
    );

    const productName = await screen.findByText(device.productName);
    expect(productName).toBeVisible();
    expect(
      screen.getByText(`${device.vendorId}:${device.productId}`)
    ).toBeVisible();
    expect(screen.getByText(device.serialNumber)).toBeVisible();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should show loading skeleton before displaying device text", async () => {
    const { asFragment } = render(
      <MockedProvider mocks={[deviceQuery(100)]}>
        <DeviceSummary deviceId="1" />
      </MockedProvider>
    );

    const before = asFragment();
    await screen.findByText(device.productName);
    const after = asFragment();

    expect(snapshotDiff(before, after)).toMatchSnapshot();
  });
});
