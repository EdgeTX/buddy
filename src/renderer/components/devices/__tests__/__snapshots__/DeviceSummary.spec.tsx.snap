// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeviceSummary /> should show details about the given device 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="ant-space ant-space-vertical"
    style="width: 100%;"
  >
    <div
      class="ant-space-item"
      style="margin-bottom: 24px;"
    >
      <div
        class="c0"
      >
        <span
          aria-label="usb"
          class="anticon anticon-usb"
          role="img"
          style="font-size: 60px; margin-bottom: 8px;"
        >
          <svg
            aria-hidden="true"
            data-icon="usb"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M760 432V144c0-17.7-14.3-32-32-32H296c-17.7 0-32 14.3-32 32v288c-66.2 0-120 52.1-120 116v356c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V548c0-24.3 21.6-44 48.1-44h495.8c26.5 0 48.1 19.7 48.1 44v356c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V548c0-63.9-53.8-116-120-116zm-424 0V184h352v248H336zm120-184h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm160 0h-48c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="c0"
      >
        <h5
          class="ant-typography"
          style="text-align: center;"
        >
          STEM32 Bootloader 2
        </h5>
        <div>
          <span
            class="ant-typography ant-typography-secondary"
            style="text-align: center;"
          >
            375897983475
          </span>
        </div>
        <div>
          <span
            class="ant-typography ant-typography-secondary"
            style="text-align: center;"
          >
            0x3453:0x4358
          </span>
        </div>
      </div>
    </div>
  </div>
  )
</DocumentFragment>
`;

exports[`<DeviceSummary /> should show loading skeleton before displaying device text 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -34,27 +34,33 @@
      </div>
      <div
        class=\\"ant-space-item\\"
      >
        <div
-         class=\\"ant-skeleton ant-skeleton-active\\"
+         class=\\"sc-bdvvtL crZnOr\\"
        >
-         <div
-           class=\\"ant-skeleton-content\\"
+         <h5
+           class=\\"ant-typography\\"
+           style=\\"text-align: center;\\"
          >
-           <h3
-             class=\\"ant-skeleton-title\\"
-             style=\\"width: 38%;\\"
-           />
-           <ul
-             class=\\"ant-skeleton-paragraph\\"
+           STEM32 Bootloader 2
+         </h5>
+         <div>
+           <span
+             class=\\"ant-typography ant-typography-secondary\\"
+             style=\\"text-align: center;\\"
            >
-             <li />
-             <li
-               style=\\"width: 61%;\\"
-             />
-           </ul>
+             375897983475
+           </span>
+         </div>
+         <div>
+           <span
+             class=\\"ant-typography ant-typography-secondary\\"
+             style=\\"text-align: center;\\"
+           >
+             0x3453:0x4358
+           </span>
          </div>
        </div>
      </div>
    </div>
    )"
`;
